src/common/page-click.service.ts

```diff
@@ -0,0 +1,19 @@
+import {Injectable, EventEmitter} from 'angular2/core';
+
+@Injectable()
+export class NglPageClickService {
+  private clickEmitter: EventEmitter<any> = new EventEmitter<any>();
+
+  constructor() {
+    document.querySelector('html').addEventListener('click', (event: any) => this.clickEmitter.next(event), true);
+  }
+
+  subscribe(callback: (_: any) => void, filter: (_: any) => boolean) {
```

Comment:
@esarbanis, nice catch with the `(document: click)` host binding. It has been pushed.